recipes/full_finetune_distributed.py

```diff
@@ -614,6 +671,60 @@ def _setup_data(
 
         return sampler, dataloader
 
+    def save_checkpoint_async(
```

Comment:
Would the idea be to eventually use async checkpoint everywhere and consolidate the files at the end? Or will we always do the all gather at the end?

Reply:
By end, I assume you mean the final checkpoint at the end of the fine-tuning?

Final checkpoint format should be the one that user configures in their job config. Intermediate checkpoints are only for fault tolerance so we can choose the Checkpointer that saves the wasted training time and GPUs the best, which in this case, is DCP (DistributedCheckpointing) with Async save. We will however provide DCP option as another Checkpointer and users can choose to get the final checkpoint in that format as well but we wont enforce it, probably. Use case I can think of is when users train using TorchTitan, where DCP is the default Checkpointer, they can then plug it into TorchTune directly for fine-tuning instead of conversions to torch.save or HF formats.